airbyte-integrations/connectors/source-sftp/src/main/java/io/airbyte/integrations/source/sftp/SftpCommand.java

```diff
@@ -0,0 +1,125 @@
+            LOGGER.info("Working directory set to : {}", path);
+        } catch (SftpException e) {
+            LOGGER.error("Could not find path : {} : ", path, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Map<String, JsonNode> getFilesSchemas() {
+        checkIfConnected();
+        Map<String, JsonNode> fileSchemas = new HashMap<>();
+        Set<String> fileNames = getFileNames();
+        LOGGER.info("Found file for sync : {}", fileNames);
+        fileNames.forEach(fileName -> {
+            ByteArrayInputStream file = client.getFile(fileName);
+            JsonNode parsedFile = tryGetFirstNode(file, fileName);
+            JsonNode schema = tryGetSchema(parsedFile, fileName);
+            fileSchemas.put(fileName, schema);
+        });
+        return fileSchemas;
+    }
+
+    private Set<String> getFileNames() {
+        checkIfConnected();
+        Vector<ChannelSftp.LsEntry> entries = new Vector<>();
+        for (SupportedFileExtension fileExtension : selectedFileExtensions) {
+            entries.addAll(client.lsFile(fileExtension));
+        }
+        return entries.stream()
+                .map(ChannelSftp.LsEntry::getFilename)
+                .filter(filePattern.asPredicate())
+                .collect(Collectors.toSet());
+    }
+
+    private JsonNode tryGetSchema(JsonNode jsonNode, String fileName) {
+        try {
+            return JsonSchemaGenerator.getJsonSchema(jsonNode);
+        } catch (Exception e) {
+            LOGGER.error("Exception occurred while trying to parse schema of file {} : ", fileName, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public JsonNode tryGetFirstNode(ByteArrayInputStream file, String fileName) {
+        try {
+            String extension = FilenameUtils.getExtension(fileName);
+            SftpFileParser parser = SftpFileParserFactory.createInstance(transformFileExtension(extension));
+            return parser.parseFileFirstEntity(file);
+        } catch (Exception e) {
+            LOGGER.error("Exception occurred while trying to parse file {} : ", fileName, e);
+            throw new RuntimeException();
+        }
+    }
+
+    public List<JsonNode> getFileData(String fileName) {
+        ByteArrayInputStream file = client.getFile(fileName);
+        return tryParseFile(file, fileName);
+    }
+
+    private List<JsonNode> tryParseFile(ByteArrayInputStream file, String fileName) {
+        try {
+            String extension = FilenameUtils.getExtension(fileName);
+            SftpFileParser parser = SftpFileParserFactory.createInstance(transformFileExtension(extension));
+            return parser.parseFile(file);
+        } catch (Exception e) {
+            LOGGER.error("Exception occurred while trying to parse file {} : ", fileName, e);
+            throw new RuntimeException();
+        }
+    }
+
+    private void checkIfConnected() {
+        if (!client.isConnected()) {
```

Comment:
Would it make sense to try and re-connect in case the client got disconnected by server for some reason?